include/linux/kfence.h

```diff
@@ -50,6 +38,24 @@ static inline bool is_kfence_addr(void *addr)
 	return unlikely((char *)addr >= __kfence_pool_start && (char *)addr < __kfence_pool_end());
 }
 
+void *__kfence_alloc(struct kmem_cache *s, size_t size, gfp_t flags);
+
+// TODO(elver): Add API doc.
+static __always_inline void *kfence_alloc(struct kmem_cache *s, size_t size, gfp_t flags)
+{
+	return static_branch_unlikely(&kfence_allocation_key) ? __kfence_alloc(s, size, flags) :
+								      NULL;
+}
+
+bool __kfence_free(void *addr);
+
+static __always_inline bool kfence_free(void *addr)
+{
+	if (!is_kfence_addr(addr))
+		return false;
+	return __kfence_free(addr);
+}
+
```

Comment:
I think you tried to reorder to make consistent with what's in the '#else ' (noops), right? Or another reason this moved?

kfence_ksize is still in the wrong place in that case relative to these. I'd probably also move kfence_free after kfence_alloc (here and below in #else).

Reply:
No, the reason was to put kfence_free() after is_kfence_addr() :)
I've reordered the functions in the noop section for consistency.